src/query/graphite/native/aggregation_functions.go

```diff
@@ -135,6 +135,39 @@ func divideSeries(ctx *common.Context, dividendSeriesList, divisorSeriesList sin
 	return r, nil
 }
 
+// aggregate takes a list of series and returns a new series containing the
+// value aggregated across the series at each datapoint using the specified function.
+func aggregate(ctx *common.Context, series singlePathSpec, fname string) (ts.SeriesList, error) {
+	switch fname {
+	case Empty, Sum, SumSeries, Total:
+		return sumSeries(ctx, multiplePathSpecs(series))
+	case Min, MinSeries:
+		return minSeries(ctx, multiplePathSpecs(series))
+	case Max, MaxSeries:
+		return maxSeries(ctx, multiplePathSpecs(series))
+	case Avg, Average, AverageSeries:
+		return averageSeries(ctx, multiplePathSpecs(series))
+	case Multiply, MultiplySeries:
+		return multiplySeries(ctx, multiplePathSpecs(series))
+	case Diff, DiffSeries:
+		return diffSeries(ctx, multiplePathSpecs(series))
+	case Count, CountSeries:
+		return countSeries(ctx, multiplePathSpecs(series))
+	case Range, RangeOf, RangeOfSeries:
+		return rangeOfSeries(ctx, series)
+	case Last, KeepLastValue:
```

Comment:
nit: Should we add "current" as alias for this too? Since it seems due to graphite documentation that "current" is an alias of "last"?
> This function can be used with aggregation functions average (or avg), avg_zero, median, sum (or total), min, max, diff, stddev, count, range (or rangeOf) , multiply & last (or current).

Reply:
I think "keepLastValue" actually shouldn't be part of the case statement here, "last" and "current" have different implementations completely to "keepLastValue" which is per timeseries (not across timeseries).

Reply:
sg, I will take another look.

Reply:
This has been updated. `KeepLastValue` is no longer part of the case statement, and `Current` is now an alias for last.